### ISSUE TITLE
The relationship is not added at creation

### ISSUE THREAD

**Issue by dorin-niscu** [laravel/scout#378]
PHP: 7.3
Laravel: 5.8
Laravel Scout: 7.1
Algolia Scout extended: 1.6

```
class Page extends Model
{   
  use Searchable;

  public function toSearchableArray()
  {
    $array = [
      'title' => optional($this->content)->title,
      'extra' => $this->extra,
    ];

    return $array;
  }

  public function content()
  {
    return $this->morphOne(Content::class, 'contentable');
  }
   
  protected $fillable = ['extra'];

  protected $with = ['content'];
}

class Content extends Model
{
    public function contentable()
    {
      return $this->morphTo();
    }

    protected $fillable = ['title'];
   
    protected $touches = ['contentable'];
}

class PageController extends Controller
{
  public function store(Request $request)
  {
    $page = Page::create($request->all());

    $page->content()->create($request->all());
   
    return $page;
  }

  public function update(Request $request, $id)
  {
    $page = $this->findOrFail($id);

    $page->update($request->all());

    $page->content->update($request->all());

    $page = $page->fresh();

    return $page;
  }
}

```
Scout import of existing instances works fine, all instances are in sync.

On update the sync works fine, $touches does the job.

```
$array = [
  'title' => 'Title',
  'extra' => 'Extra field',
];

```
**_On store the content relation is null and not updated when is created._**

```
$array = [
  'title' => null,
  'extra' => 'Extra field',
];
```

**Comment by driesvints** [comment]
I'm sorry but we can't provide support for Algolia Scout extended, only Scout itself. Is the problem specifically happening for Algolia Scout extended? If so it's best that you post this on their issue tracker. If not: please test without Algolia Scout extended installed to see if it also occurs.

**Comment by dorin-niscu** [comment]
@driesvints I confirm that I removed Algolia extended and it is the same behaviour. It works fine on update but not on store.

**Comment by driesvints** [comment]
@dorin-niscu I have to admit that I'm a bit swamped atm so I don't have time to deep-dive into this, sorry. 

Maybe @nunomaduro or someone else can help out here?

**Comment by diegoaurino** [comment]
I was facing a similar problem these days (with Algolia). The problem is that, for some reason, Scout appears to ignore relationship synchronizations on the store method. It looks like the `sync()` method does not change (touches) the timestamp of the model object that has been created on the store method in question. 

[$touches](https://laravel.com/docs/5.8/eloquent-relationships#touching-parent-timestamps) looks to be a solution for those that are not syncing or attaching, but creating or updating the synced objects. 

A small workaround, I did, was to `touch()` the object, what will update its timestamp. Then Scout will pick the relationship too. But you cannot do that on the store method; it will happen so quickly that the timestamp will not change. 

A workaround is to add an event and a listener for the store method. Then, you only need to use the last one to trigger a job that will touch your model object. The job should be delayed at least 1 (one) minute. That appears to be enough for Scout to pick the new timestamp and index the object with the relationship. That approach does not interfere with the user experience because the job will be handled behind the scene. 

In other words, it does not look like a Scout bug, but only a thing to be aware of. 

Hope this helps someone!

**Comment by driesvints** [comment]
Going to close this one. Please see the answer above by @diegoaurino, thanks!
